spec.html

```diff
@@ -22007,7 +22007,7 @@ <h1>ParseModule ( _sourceText_, _realm_, _hostDefined_ )</h1>
         </emu-clause>
 
         <emu-clause id="sec-getexportednames">
-          <h1>GetExportedNames ( _exportStarSet_ ) Concrete Method</h1>
+          <h1>GetExportedNames ( _exportStarSet_, _nsModule_ ) Concrete Method</h1>
```

Comment:
It’d be nice if there was some sort of assertion on the type of the arguments here - for both exportStarSet and nsModule.

Reply:
Sure, I can add that.

Reply:
Added.

Reply:
Should `nsModule` be passed into this function for the base (non-recursive) case?

Reply:
The way Dynamic Modules are written, both cases are treated the same - the module tracks its own namespace equally to any other namespace which reflects its own export names, there are no conditionals in this handling at all.

Reply:
In https://guybedford.github.io/proposal-dynamic-modules/#sec-getmodulenamespace, `module` is passed to GetExportedNames. Did you intend for the same change here?

Reply:
Yes, thanks for spotting that!